pkg/gofr/datasource/surrealdb/surrealdb.go

```diff
@@ -0,0 +1,634 @@
+}
+
+// Insert inserts a new record into the specified table in SurrealDB.
+func (c *Client) Insert(ctx context.Context, table string, data any) (*Response, error) {
+	query := fmt.Sprintf("INSERT INTO %s", table)
+	span := c.addTrace(ctx, "Insert", query)
+	if span != nil {
+		defer span.End()
+	}
+
+	if c.db == nil {
+		return nil, errNotConnected
+	}
+
+	startTime := time.Now()
+
+	defer c.sendOperationStats(&QueryLog{
+		Query:         query,
+		OperationName: "insert",
+		Namespace:     c.config.Namespace,
+		Database:      c.config.Database,
+		Collection:    table,
+		Data:          data,
+		Span:          span,
+	}, startTime)
+
+	var insertResult Response
+	if err := c.db.Send(&insertResult, "insert", table, data); err != nil {
+		return nil, err
+	}
+
+	c.metrics.RecordHistogram(ctx, "surreal_db_operation_duration", 0, "operation", "insert")
+
+	return &insertResult, nil
+}
+
+// Delete removes a record from the specified table in SurrealDB.
+func (c *Client) Delete(ctx context.Context, table, id string) (any, error) {
+	query := fmt.Sprintf("DELETE FROM %s WHERE id = %s", table, id)
+	span := c.addTrace(ctx, "Delete", query)
+	if span != nil {
+		defer span.End()
+	}
+
+	if c.db == nil {
+		return nil, errNotConnected
+	}
+
+	startTime := time.Now()
+
+	defer c.sendOperationStats(&QueryLog{
+		Query:         query,
+		OperationName: "delete",
+		Namespace:     c.config.Namespace,
+		Database:      c.config.Database,
+		Collection:    table,
+		ID:            id,
+		Span:          span,
+	}, startTime)
+
+	var DeleteResult Response
+
+	arg := models.RecordID{
+		Table: table,
+		ID:    id,
+	}
+
+	if err := c.db.Send(&DeleteResult, "delete", arg); err != nil {
+		return nil, err
+	}
+
+	c.metrics.RecordHistogram(ctx, "surreal_db_operation_duration", 0, "operation", "delete")
+
+	return DeleteResult.Result, nil
+}
+
+// addTrace starts a new trace span for the specified method and query.
+func (c *Client) addTrace(ctx context.Context, method, query string) trace.Span {
+	if c.tracer == nil {
+		return nil
+	}
+
+	_, span := c.tracer.Start(ctx, fmt.Sprintf("SurrealDB.%v", method))
+	span.SetAttributes(
+		attribute.String("surrealdb.query", query),
+		attribute.String("surrealdb.namespace", c.config.Namespace),
+		attribute.String("surrealdb.database", c.config.Database),
+	)
+	return span
+}
+
+func (c *Client) sendOperationStats(ql *QueryLog, startTime time.Time) {
+	duration := time.Since(startTime).Microseconds()
+
+	ql.Duration = duration
+
+	c.logger.Debug(ql)
+
+	ql.Namespace = c.config.Namespace
+	ql.Database = c.config.Database
+
+	if ql.Span != nil {
+		ql.Span.SetAttributes(
+			attribute.Int64("surrealdb.duration", duration),
+			attribute.String("surrealdb.query", ql.Query),
+			attribute.String("surrealdb.operation", ql.OperationName),
+			attribute.String("surrealdb.namespace", ql.Namespace),
+			attribute.String("surrealdb.database", ql.Database),
+			attribute.String("surrealdb.collection", ql.Collection),
+		)
+	}
```

Comment:
```suggestion
func (c *Client) sendOperationStats(ql *QueryLog, startTime time.Time) {
	duration := time.Since(startTime).Microseconds()

	ql.Duration = duration

	c.logger.Debug(ql)

	ql.Namespace = c.config.Namespace
	ql.Database = c.config.Database

	if ql.Span == nil {
		return
	}

	ql.Span.SetAttributes(
		attribute.Int64("surrealdb.duration", duration),
		attribute.String("surrealdb.query", ql.Query),
		attribute.String("surrealdb.operation", ql.OperationName),
		attribute.String("surrealdb.namespace", ql.Namespace),
		attribute.String("surrealdb.database", ql.Database),
		attribute.String("surrealdb.collection", ql.Collection),
	)
```